Fix documentation in `cardano-balance-tx`.

jonathanknowles opened the pull request:
## Issue

ADP-3186

## Description

This PR fixes the documentation for selected functions in `cardano-balance-tx`.

In particular, references to "the wallet" have been removed.

jonathanknowles committed:
Fix documentation for `maximumLengthChangeAddress`.

jonathanknowles committed:
Fix documentation for `assignChangeAddresses`.

jonathanknowles committed:
Fix documentation for `toRecentEra`.

Anviking commented on lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs:
```diff
@@ -1170,8 +1170,7 @@ selectAssets era (ProtocolParameters pp) utxoAssumptions outs redeemers
 
-    -- | Returns the longest address that the wallet can generate for a given
-    --   key.
+    -- | Returns a dummy change address of the maximum possible length.
```
Could be good to write: "maximum possible length **'getChangeAddressGen' can generate**" or something similar, as maximum possible length isn't a global property

jonathanknowles replied:
@Anviking good point!

I've added a statement of this property in the following commit: 212301f0315e5c15606316f82f5e7cdcbf8f4ee0.

jonathanknowles committed:
Add property to documentation of `maxLengthChangeAddress`.

In response to review feedback:

https://github.com/cardano-foundation/cardano-wallet/pull/4234#discussion_r1395402817

jonathanknowles committed:
Add documentation for `getChangeAddressGen`.

jonathanknowles committed:
Shorten name of field to `genChangeAddress`.

This is (arguably) more direct, and still consistent with the purpose of
this function, which is to generate a new change address.
